modules/order/src/Form/UseMultiProfileConfirmForm.php

```diff
@@ -0,0 +1,390 @@
+      $container->get('current_route_match'),
+      $container->get('config.storage'),
+      $container->get('entity_type.manager'),
+      $container->get('entity_field.manager'),
+      $container->get('commerce.configurable_field_manager')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getDescription() {
+    // Check if there are existing orders on this order type.
+    $description = $this->getExistingOrderCountDescription();
+
+    $description .= '<strong>'
+      . $this->t('This action cannot be undone. You cannot switch back to
+       using the single profile again.')
+      . '</strong>';
+
+    return $description;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() : string {
+    return $this->entity->getEntityTypeId() . '_confirm_form';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getQuestion() {
+    return $this->t('Are you sure you want to switch to using multiple
+      profiles for shipping and billing for the %label order type?', [
+        '%label' => $this->entity->label(),
+      ]
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getConfirmText() {
+    return $this->t('Switch to Using Multiple Profiles');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getCancelUrl() {
+    return $this->entity->toUrl('collection');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    // Check if we have incompatible module versions as modules like Commerce
+    // Shipping and Commerce POS will be affected when we switch to split
+    // profiles.
+    $incompatible_modules = $this->getIncompatibleModules();
+
+    // If we have modules that are running incompatible versions, output a
+    // warning message to the user.
+    if ($incompatible_modules) {
+      $this->messenger()->addWarning($this->t('The following modules are
+        running incompatible versions for the switch to split profiles and it
+        could possibly render the site as unusable.
+        <p><strong>@modules</strong></p>
+        <p>Please upgrade and try again.</p>', [
+          '@modules' => implode('<br>', $incompatible_modules)
+        ]
+      ));
+
+      return;
+    }
+
+    return parent::buildForm($form, $form_state);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state) {
+    /** @var \Drupal\commerce_order\Entity\OrderTypeInterface $order_type */
+    $order_type = $this->entity;
+
+    // Create the billing and shipping profile types.
+    $this->createBillingShippingProfiles();
+
+    // Migrate any fields added to the customer profile type to the billing and
+    // shipping profile types.
+    $this->migrateProfileFields();
+
+    // Batch process to migrate the existing order profiles from the customer
+    // profile type to use the billing/shipping profile types.
+    $this->migrateExistingProfiles();
```

Comment:
**CHANGE REQUEST**

This is confusing, we're using the word "profiles" equally whether we're referring to profiles or to profile types. I suggest the following:

createProfileTypes()
migrateProfileTypeFields()
migrateExistingProfiles()